contributor-docs/adrs/adr-005.md

```diff
@@ -0,0 +1,65 @@
+# ADR 5: Migration plan for deprecated components
+
+## Status
+
+Proposed
+
+## Context
+
+As we work on maturity of our components, we will sometimes need to deprecate components that are replaced with new components.
+
+## Here are the 3 proposed stages:
+
+### Stage 1
+
+Start new component outside the main bundle. Folks who want to try it, need to explicitly import it from the `drafts` bundle.
+
+`import { ActionMenu} from '@primer/react/drafts'`
+
+The contract with consumers is - you are opting into a rewrite of the old component that might not cover all the cases of the old component yet. If you are using both the old and new version of the component in different pages, you are paying some additional bundlesize cost.
+
+Note: If it's a 1:1 replacement, it's useful to keep the same name.
```

Comment:
So we would stop adding "2" as a suffix? For example: `TextInput2` would just be called `TextInput` if it was a 1-to-1 replacement.

Reply:
I think it's a mixed bag,

In the shipped bundle:  `import { TextInput } from '@primer/react/labs'`

but the file name might still be `TextInput2.tsx` and the docs would still say `TextInput v2`. Would be nice to have some consistency around this.

Reply:
☝️  Could you add that to this doc for clarity?

Reply:
Done :)